layout-server/src/main/java/org/finos/vuu/layoutserver/model/Layout.java

```diff
@@ -0,0 +1,29 @@
+package org.finos.vuu.layoutserver.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.UUID;
+
+@Data
+@NoArgsConstructor
+@Entity
+public class Layout {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.UUID)
+    private UUID id;
```

Comment:
How does this work when we do a create? Do we end up with one unique ID that matches a Layout to its Metadata, or does it generate a different one for each and match them some other way?

Reply:
Layout has a unique generated ID, and Metadata has a unique generated ID. The `@OneToOne` annotation creates the foreign key to relate these entities.

https://www.baeldung.com/jpa-join-column#oneToOne_mapping